### ISSUE TITLE
Make the Wave Shoot Waves?

### ISSUE THREAD

**Issue by genNAowl** [Anuken/Mindustry-Suggestions#1708]
Just gonna put this crappy idea here: 

**What?**

Make the wave shoot waves of liquid

**Why?**

The name is wave not stream

**Yummy Xs**

1. - [X] Not Suggested?
2. - [X] Not Planned?
3. - [X] Not Implemented?
4. - [X] Not Discouraged?

**Comment by Yell245** [comment]
wh
How'd you make it an actual wave????

**Comment by itcannotbe** [comment]
Actually my suggestion for a T2 wave is something that would be able to shoot a "wave" of liquid not just dots.

**Comment by iLagSoUhax** [comment]
I think he means 'wave' as in short pulses of the liquid it is fed, similar to a sector of a radar pulse. I am not against the idea, it might provide variation between Tsunami and Wave, but implementing the same amount of knockback might be difficult.

**Comment by Humanoid-X** [comment]
lololol good idea

**Comment by joshuaptfan** [comment]
How would a "wave" work? Would it be an arc (section of a circle) of dots fired in a cone from the wave? Because that's the only way the "wave" could hit multiple enemies with this new wider wave, for the wave to be made up of individual projectiles. But doing it this way would mean that enemies that are close to the wave would get hit with more projectiles, meaning that the amount of knockback would vary greatly depending on distance.

Also, a periodic arc of dots would look much worse than the current implementation.

**Comment by genNAowl** [comment]
> How would a "wave" work

Thats up for the implementer to decide; I just want some sort of a wave

**Comment by Kyllingene** [comment]
genNAowl, suggesting things without at least seeing if there is a feasible way to implement them is a bad practice, as discussions like these arise; nothing helpful because there is no real way to implement this. It would be cool if it could happen, but since it can't (as far as we've seen), it's not a helpful suggestion.

**Comment by genNAowl** [comment]
@Kyllingene 

Off the top of my head, here are two ways of implementing such a thing: 

Like joshuaptfan said, a bunch of bullets: 

![Wave Shot1](https://user-images.githubusercontent.com/68400583/103493679-9525d200-4de7-11eb-910f-e9e75d75cf60.gif)

Or a special "liquid wave" bullet: 

![Wave Shot2](https://user-images.githubusercontent.com/68400583/103502337-c90ff000-4e05-11eb-826b-c01bfe3d95d3.gif)

There are definitely more ways to implement this, but I think these examples are more than enough to show that **there is a feasible way to implement this.**

The bottom line is that you need to **get your facts right** before criticizing others.

**Comment by Kyllingene** [comment]
By feasible, I mean that the multi-bullet solution needs to be pretty well thought-out to work, and same with the single-bullet strategy. Since the OP didn't suggest any solutions, this is now a forum for making entirely new mechanics, which I do not believe this is supposed to be.

**Comment by itcannotbe** [comment]
Actually this is a better place for this than discord.

**Comment by QmelZ** [comment]
> Actually this is a better place for this than discord.

i agree.


> ![Wave Shot1](https://user-images.githubusercontent.com/68400583/103493679-9525d200-4de7-11eb-910f-e9e75d75cf60.gif)

this multiple bullet solution seems better, but needs balancing because it gets wider the bullet travels farther

**Comment by itcannotbe** [comment]
Stupid question though. How can I fight fires now?

**Comment by QmelZ** [comment]
the water still will put out the fire

**Comment by itcannotbe** [comment]
Doesn't it takes some seconds of the current wave shot to.kill the fire?

**Comment by QmelZ** [comment]
well fire might also be nerfed depending on the situation, as i said before, this is an actually decent suggestion and has a lot to discuss and balance

**Comment by itcannotbe** [comment]
And that's what we're doing.

**Comment by SavageCreeper64** [comment]
Sure, I really see no reason not to have this. Tsunami is really just a better wave, so in my opinion, wave is really just useless for its purpose and making the wave shoot actual waves would more practical and honestly, I don't think anyone would care if the wave was changed.
